pkg/types/adopted_resource_reconciler.go

```diff
@@ -0,0 +1,40 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package types
+
+import (
+	"context"
+
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+)
+
+// AdoptedResourceReconciler is responsible for reconciling an adopted resource
+// that represent AWS service API resource.
+// It implements the upstream controller-runtime `Reconciler`
+// interface.
+type AdoptedResourceReconciler interface {
```

Comment:
I think this interface should remain in the `adoption_reconciler.go` file.

Reply:
I just placed it in same directory as AWSResourceReconciler, because I think these two belong together as top level type in same directory(`types`).
Can you provide reasoning for your suggestion? (Do you think AdoptedResourceReconciler is not a first class type in ACK runtime, unlike AWSResourceReconciler ?)

Reply:
Ohhhh yeah I see what you mean. Great